types/hapi/definitions/plugin/plugin.d.ts

```diff
@@ -0,0 +1,57 @@
+import {Server, ServerRegisterOptions} from "hapi";
+
+export interface PluginsStates {
+    [pluginName: string]: any;
+}
+
+export interface PluginSpecificConfiguration {
+
+}
+
+export interface PluginNameVersion {
+    /**
+     * (required) the plugin name string. The name is used as a unique key. Published plugins (e.g. published in the npm
+     * registry) should use the same name as the name field in their 'package.json' file. Names must be
+     * unique within each application.
+     */
+    name: string;
+
+    /** optional plugin version. The version is only used informatively to enable other plugins to find out the versions loaded. The version should be the same as the one specified in the plugin's 'package.json' file. */
+    version?: string;
+}
+
+export interface PluginPackage {
+
+    /**
+     * Alternatively, the name and version can be included via the pkg property containing the 'package.json' file for the module which already has the name and version included
+     */
+    pkg: any;
```

Comment:
Weren't there some basic requirements such as name and version? Maybe we can narrow this a bit.